https://github.com/wbx-github/uclibc-ng/commit/74c69da52ad5029cd2fc2d3d41d581022ed0a7aa
fix static linking of pthread apps
When compiling python you get duplicate symbol problem.
Seen in the autobuilders of buildroot project.


--- a/libc/sysdeps/linux/common/setregid.c
+++ b/libc/sysdeps/linux/common/setregid.c
@@ -37,4 +37,4 @@ int setregid(gid_t rgid, gid_t egid)
 }
 #endif
 
-libc_hidden_def(setregid)
+libc_hidden_weak(setregid)
--- a/libc/sysdeps/linux/common/setresgid.c
+++ b/libc/sysdeps/linux/common/setresgid.c
@@ -16,7 +16,7 @@
 # define __NR_setresgid __NR_setresgid32
 
 _syscall3(int, setresgid, gid_t, rgid, gid_t, egid, gid_t, sgid)
-libc_hidden_def(setresgid)
+libc_hidden_weak(setresgid)
 
 #elif defined(__NR_setresgid)
 
@@ -34,7 +34,7 @@ int setresgid(gid_t rgid, gid_t egid, gi
 	}
 	return (__syscall_setresgid(rgid, egid, sgid));
 }
-libc_hidden_def(setresgid)
+libc_hidden_weak(setresgid)
 
 #endif
 
--- a/libc/sysdeps/linux/common/setresuid.c
+++ b/libc/sysdeps/linux/common/setresuid.c
@@ -16,7 +16,7 @@
 # define __NR_setresuid __NR_setresuid32
 
 _syscall3(int, setresuid, uid_t, ruid, uid_t, euid, uid_t, suid)
-libc_hidden_def(setresuid)
+libc_hidden_weak(setresuid)
 
 #elif defined(__NR_setresuid)
 
@@ -34,7 +34,7 @@ int setresuid(uid_t ruid, uid_t euid, ui
 	}
 	return (__syscall_setresuid(ruid, euid, suid));
 }
-libc_hidden_def(setresuid)
+libc_hidden_weak(setresuid)
 
 #endif
 
--- a/libc/sysdeps/linux/common/setreuid.c
+++ b/libc/sysdeps/linux/common/setreuid.c
@@ -37,4 +37,4 @@ int setreuid(uid_t ruid, uid_t euid)
 }
 #endif
 
-libc_hidden_def(setreuid)
+libc_hidden_weak(setreuid)
--- a/libc/sysdeps/linux/common/seteuid.c
+++ b/libc/sysdeps/linux/common/seteuid.c
@@ -36,4 +36,4 @@ int seteuid(uid_t uid)
 
     return result;
 }
-libc_hidden_def(seteuid)
+libc_hidden_weak(seteuid)
